### PR TITLE
fixed #21848: While instantiating a sound flag, popup persists on undo

### LINKED ISSUE #21848: While instantiating a sound flag, popup persists on undo
These two issues occur when you undo while instantiating a sound flag for the first time.

### **After selecting a single sound**

https://github.com/musescore/MuseScore/assets/20806406/bdbf8345-8356-44d7-9780-184b0efe69a6

1. Add text via Cmd + T
2. Click the sound flag button
3. Click one button to change the sound
4. Undo
5. Sound flag button disappears, but the popup persists

**Expected behavior:**
On undo, popup should close

### **After multiple sounds have been selected**

https://github.com/musescore/MuseScore/assets/20806406/11e654f5-7fd0-49f1-a067-d823dcd4e426

1. Add text via Cmd + T
2. Click the sound flag button
3. Change the sound 3 or more times
4. Undo twice
5. The sound flag gets removed on the second undo and the popup persists

**Expected behavior:**
Undo goes back through each sound that was selected, and once it gets to the first sound, undo removes the sound flag and closes the popup

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #21848

### CODE REVIEW

**Comment by avvvvve** [comment]
@Eism Still seeing the first issue on certain sound flags—the popup persists after a single change and undo. It only appears to happen when the sound flag does not have any buttons accented upon opening it for the first time.

https://github.com/musescore/MuseScore/assets/20806406/69dbc800-100c-4269-a1c9-856d040ea3e4

Attaching the score where I'm seeing this. It's happening at the first measure of Trumpet 1, but interestingly is working properly in the second measure of Pipe Organ.

[Sound Flags testing.mscz.zip](https://github.com/musescore/MuseScore/files/14574369/Sound.Flags.testing.mscz.zip)

**Comment by Eism** [comment]
@avvvvve fixed

File: src/notation/view/abstractelementpopupmodel.cpp [hunk 1]
```diff
@@ -190,6 +199,11 @@ mu::engraving::ElementType AbstractElementPopupModel::elementType() const
     return ELEMENT_POPUP_TYPES.key(m_modelType, ElementType::INVALID);
 }
 
+mu::engraving::ElementTypeSet AbstractElementPopupModel::dependentElementTypes() const
```
Comment (RomanPudashkin) [inline src/notation/view/abstractelementpopupmodel.cpp:202]: const mu::engraving::ElementTypeSet&

File: src/engraving/dom/edit.cpp [hunk 2]
```diff
@@ -3484,6 +3484,8 @@ void Score::cmdDeleteSelection()
                         break;
                     }
                 }
+            } else if (e->isSoundFlag()) {
+                tick = toSoundFlag(e)->parentItem()->tick();
```
Comment (RomanPudashkin) [inline src/engraving/dom/edit.cpp:3488]: It looks like we can use the SoundFlag tick directly: 

tick = toSoundFlag(e)->tick()

File: src/notation/view/abstractelementpopupmodel.cpp [hunk 3]
```diff
@@ -32,6 +32,13 @@ static const QMap<mu::engraving::ElementType, PopupModelType> ELEMENT_POPUP_TYPE
     { mu::engraving::ElementType::SOUND_FLAG, PopupModelType::TYPE_SOUND_FLAG },
 };
 
+static const QMap<PopupModelType, mu::engraving::ElementTypeSet> POPUP_DEPENDENT_ELEMENT_TYPES = {
```
Comment (RomanPudashkin) [inline src/notation/view/abstractelementpopupmodel.cpp:35]: QHash looks better here

**Review by RomanPudashkin — APPROVED** [review]

**Comment by avvvvve** [comment]
@Eism Love it, works great now. Thanks!

**Pull request merged**